camera.c

```diff
@@ -0,0 +1,298 @@
+/*
+ * Copyright (c) 2024, Linaro Ltd.
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+#include <errno.h>
+#include <fcntl.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/ioctl.h>
+#include <sys/mman.h>
+#include <unistd.h>
+
+#include <linux/videodev2.h>
+
+#include <jpeglib.h>
+
+#include "camera.h"
+#include "device.h"
+
+int init_device(const char *dev_name, int *fd_ref, char **driver_ref)
```

Comment:
Please prefix all functions with `camera_` prefix. Use `static` for all functions that are no exported to other modules.